File: trees/btree_traversal.c
#include<stdio.h>
#include<stdlib.h>

typedef struct  _Node
{
    int data;
    struct _Node *left;
    struct _Node *right;
    
}Node;

Node* newNode(int data)
{
    Node *ptr = (Node*) malloc(sizeof(Node));
    ptr->data = data;    
    ptr->left = NULL;
    ptr->right = NULL;

    return ptr;
}

int inorder_travarsal(Node *root)
{
    if(root == NULL)
        return 0;
    inorder_travarsal(root->left);
    printf("%d ",root->data);
    inorder_travarsal(root->right);

}

int preorder_travarsal(Node *root)
{
    if(root == NULL)
        return 0;
    printf("%d ",root->data);
    preorder_travarsal(root->left);
    preorder_travarsal(root->right);

}

int postorder_travarsal(Node *root)
{
    if(root == NULL)
        return 0;   
    postorder_travarsal(root->left);
    postorder_travarsal(root->right);
    printf("%d ",root->data);

}


int main()
{
    Node *root = newNode(2);
    root->left = newNode(4);
    root->right = newNode(6);

    root->left->left = newNode(8);
    root->left->right = newNode(10);

    Node *ref = root;
    printf("in-order traversal\n");
    inorder_travarsal(ref);
    printf("\npre-order traversal\n");  
    preorder_travarsal(ref);
    printf("\npost-order traversal\n");
    postorder_travarsal(ref);
    return 0;
}

File: trees/btree_traversal.py
class Btree:
    def __init__(self,data):
        self.data = data
        self.left = None
        self.right = None
        
    def inorder_traversal(self):
        if self.left:
            self.left.inorder_traversal()
        print(" ",self.data)
        if self.right:
            self.right.inorder_traversal()
    def preorder_traversal(self):
        print(self.data)
        if self.left:
            self.left.preorder_traversal()
        if self.right:
            self.right.preorder_traversal()
    def postorder_traversal(self):
        if self.left:
            self.left.postorder_traversal()
        if self.right:
            self.right.postorder_traversal()
        print(self.data)

Bt  = Btree(2) 
Bt.left = Btree(4)
Bt.right =  Btree(6)

Bt.left.left = Btree(8)
Bt.left.right = Btree(10)

print("in-order traversal")
Bt.inorder_traversal()
print("pre-order traversal")
Bt.preorder_traversal()
print("post-order traversal")
Bt.postorder_traversal()



